beacon-chain/core/helpers/randao.go

```diff
@@ -12,27 +13,30 @@ import (
 //
 // Spec pseudocode definition:
 //   def generate_seed(state: BeaconState,
-//                  epoch: Epoch) -> Bytes32:
+// 	                   epoch: Epoch) -> Bytes32:
 //    """
 //    Generate a seed for the given ``epoch``.
 //    """
 //    return hash(
-//        get_randao_mix(state, epoch - MIN_SEED_LOOKAHED) +
-//        get_active_index_root(state, epoch)
+//               get_randao_mix(state, epoch - MIN_SEED_LOOKAHEAD) +
+//               get_active_index_root(state, epoch) +
+//               int_to_bytes32(epoch)
 //    )
 func GenerateSeed(state *pb.BeaconState, wantedEpoch uint64) ([32]byte, error) {
-	if wantedEpoch > params.BeaconConfig().MinSeedLookahead {
-		wantedEpoch -= params.BeaconConfig().MinSeedLookahead
+	if wantedEpoch < params.BeaconConfig().MinSeedLookahead {
+		return [32]byte{}, fmt.Errorf("can't generate seed for epoch: %d that is smaller then MIN_SEED_LOOKAHEAD: %d", wantedEpoch, params.BeaconConfig().MinSeedLookahead)
```

Comment:
don't return error here, it's not an `assertion` from spec

Reply:
removed